### ISSUE TITLE
Using "Any" generates incorrect rules

### ISSUE THREAD

**Issue by mm-gmbd** [firebase/bolt#111]
Consider the following setup:

```
type MyType {
  num: Number,
  str: String,
  any: Any
}

path / {
  read()  = true;
  write() = true;
}

path /path is MyType {
  read() = true();
  write() = true();
}
```

The third property of `MyType` is defined as type Any, so I would expect that, at `path` in the rules, it would ensure that `any` was included in a `newData.hasChildren([])` statement, and then the specific `.validate` rule would check to ensure that it wasn't null. However, the following rules are generated:

```
{
  "rules": {
    "path": {
      ".validate": "newData.hasChildren(['num', 'str', 'any'])",
      "num": {
        ".validate": "newData.isNumber()"
      },
      "str": {
        ".validate": "newData.isString()"
      },
      "$other": {
        ".validate": "false"
      },
      ".read": "true()",
      ".write": "true()"
    }
  }
}
```

Notice the complete lack of a `.validate` rule for `any`, even though the parent `.validate` includes it in `.hasChildren()`. Because this is left out, the `$other` wildcard `.validate: false` rule breaks the rule generation!

**Comment by mckoss** [comment]
Yes!  Same cause as issue #97.

**Comment by mm-gmbd** [comment]
Hey @mckoss - any update on this one? I just revisited some rules, used `Any` as a type, and fell into the same pitfall. Looking at the docs indicated that I wasn't doing anything wrong, but I had to come here to remember the issue - maybe the docs could be updated in the meantime (i.e. next to "Any" there could be a warning that says not to use it, or it could be removed entirely for now)?

**Comment by RomansBermans** [comment]
Hey @mckoss I am trying to rewrite the [Firebase Queue](https://github.com/firebase/firebase-queue) rules from JSON to Bolt and below is the case when type `Any` doesn't work:

```
...
type TaskErrorDetails {
  error: NonEmptyString | Null,
  error_stack: NonEmptyString | Null,
  previous_state: NonEmptyString | Null,
  original_task: Any | Null,
  attempts: GreaterThanZeroNumber | Null
}
...
```
Will generate:
```
...
"_error_details": {
  ".validate": "newData.hasChildren() || newData.val() == null",
  "error": {
    ".validate": "newData.isString() && newData.val().length > 0 || newData.val() == null"
  },
  "error_stack": {
    ".validate": "newData.isString() && newData.val().length > 0 || newData.val() == null"
  },
  "previous_state": {
    ".validate": "newData.isString() && newData.val().length > 0 || newData.val() == null"
  },
  "attempts": {
    ".validate": "newData.isNumber() && newData.val() > 0 || newData.val() == null"
  },
  "$other": {
    ".validate": "false"
  }
}
...
```
As you can see `original_task` has been removed from the generated rules.

If on the other hand I try to generate them from the following:
```
...
type TaskErrorDetails {
  error: NonEmptyString | Null,
  error_stack: NonEmptyString | Null,
  previous_state: NonEmptyString | Null,
  original_task: Any,
  attempts: GreaterThanZeroNumber | Null
}
...
```
I get:
```
...
"_error_details": {
  ".validate": "newData.hasChildren() && newData.hasChildren(['original_task']) || newData.val() == null",
  "error": {
    ".validate": "newData.isString() && newData.val().length > 0 || newData.val() == null"
  },
  "error_stack": {
    ".validate": "newData.isString() && newData.val().length > 0 || newData.val() == null"
  },
  "previous_state": {
    ".validate": "newData.isString() && newData.val().length > 0 || newData.val() == null"
  },
  "attempts": {
    ".validate": "newData.isNumber() && newData.val() > 0 || newData.val() == null"
  },
  "$other": {
    ".validate": "false"
  }
}
...
```
Again the `original_task` has been removed from the generated rules, but `.validate` references it.

**Comment by pandaiolo** [comment]
Hi

Also bumped into this issue. I use this workaround:

    // Use this instead of type `Any`
    type Anything {
        validate() {
            this.isString() || this.isNumber() || this.isBoolean() || this.hasChildren()
        }
    }

**Comment by mckoss** [comment]
Fixed by PR #166
